x/nexus/types/params.go

```diff
@@ -37,27 +46,57 @@ func (m *Params) ParamSetPairs() params.ParamSetPairs {
 		set on the correct Params data struct
 	*/
 	return params.ParamSetPairs{
-		params.NewParamSetPair(KeyChainActivationThreshold, &m.ChainActivationThreshold, validateChainActivationThreshold),
+		params.NewParamSetPair(KeyChainActivationThreshold, &m.ChainActivationThreshold, validateThresholdWith("ChainActivationThreshold")),
+		params.NewParamSetPair(KeyChainMaintainerMissingVoteThreshold, &m.ChainMaintainerMissingVoteThreshold, validateThresholdWith("ChainMaintainerMissingVoteThreshold")),
+		params.NewParamSetPair(KeyChainMaintainerIncorrectVoteThreshold, &m.ChainMaintainerIncorrectVoteThreshold, validateThresholdWith("ChainMaintainerIncorrectVoteThreshold")),
+		params.NewParamSetPair(KeyChainMaintainerCheckWindow, &m.ChainMaintainerCheckWindow, validateChainMaintainerCheckWindow),
 	}
 }
 
 // Validate checks if the parameters are valid
 func (m Params) Validate() error {
-	if err := validateChainActivationThreshold(m.ChainActivationThreshold); err != nil {
+	if err := validateThresholdWith("ChainActivationThreshold")(m.ChainActivationThreshold); err != nil {
+		return err
+	}
+
+	if err := validateThresholdWith("ChainMaintainerMissingVoteThreshold")(m.ChainMaintainerMissingVoteThreshold); err != nil {
+		return err
+	}
+
+	if err := validateThresholdWith("ChainMaintainerIncorrectVoteThreshold")(m.ChainMaintainerIncorrectVoteThreshold); err != nil {
+		return err
+	}
+
+	if err := validateChainMaintainerCheckWindow(m.ChainMaintainerCheckWindow); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func validateChainActivationThreshold(chainActivationThreshold interface{}) error {
-	val, ok := chainActivationThreshold.(utils.Threshold)
+func validateThresholdWith(paramName string) func(interface{}) error {
+	return func(i interface{}) error {
+		val, ok := i.(utils.Threshold)
+		if !ok {
+			return fmt.Errorf("invalid parameter type for %s: %T", paramName, i)
+		}
+
+		if err := val.Validate(); err != nil {
+			return sdkerrors.Wrapf(err, "invalid %s", paramName)
+		}
+
+		return nil
+	}
+}
+
+func validateChainMaintainerCheckWindow(i interface{}) error {
+	val, ok := i.(uint32)
 	if !ok {
-		return fmt.Errorf("invalid parameter type for ChainActivationThreshold: %T", chainActivationThreshold)
+		return fmt.Errorf("invalid parameter type for ChainMaintainerCheckWindow: %T", i)
 	}
 
-	if err := val.Validate(); err != nil {
-		return sdkerrors.Wrap(err, "invalid ChainActivationThreshold")
+	if val == 0 {
```

Comment:
While it would be preferable to have arbitrary length bitmaps, if we don't do that, we should check that this window is <= 256

Reply:
yes will do